modules/cookbooks/pages/rag.adoc

```diff
@@ -0,0 +1,212 @@
+= Retrieval-Augmented Generation (RAG)
+// tag::single-source[]
+:description: How to configure Redpanda Connect to create a RAG pipeline, using PostgreSQL and PGVector.
+
+ifndef::env-cloud[]
+This cookbook shows you how to create a Retrieval-Augmented Generation (RAG) pipeline, using PostgreSQL and https://github.com/pgvector/pgvector[PGVector^]. 
+
+Follow the cookbook to:
+
+* Create two Redpanda Connect data pipelines: one for creating vector embeddings with https://ollama.ai[Ollama^] and another for searching the data, using https://github.com/pgvector/pgvector[PGVector^].
+* Run the two pipelines in parallel using xref:guides:streams_mode/about.adoc[streams mode].
+endif::[]
+ifdef::env-cloud[]
+This cookbook shows you how to create a vector embeddings indexing pipeline for Retrieval-Augmented Generation (RAG), using PostgreSQL and https://github.com/pgvector/pgvector[PGVector^]. 
+
+Follow the cookbook to:
+
+* Take textual data from a Redpanda topic and compute vector embeddings for it using https://ollama.ai[Ollama^]
+* Write the pipeline output into a PostgreSQL table with a https://github.com/pgvector/pgvector[PGVector^] index on the embeddings column.
+endif::[]
+
+== Compute the embeddings
+
+ifndef::env-cloud[]
+Start by defining the indexing pipeline, which takes textual data from Redpanda, computes vector embeddings for it, and then write it into a PostgreSQL table with a PGVector index on the embeddings column.
+endif::[]
+
+[source,bash]
+----
+rpk topic create articles
+echo '{
+  "type": "article",
+  "article": {
+    "id": "123foo",
+    "title": "Dogs Stop Barking",
+    "content": "The world was shocked this morning to find that all dogs have stopped barking."
+  }
+}' | rpk topic produce acticles -f '%v'
+----
+
+Your indexing pipeline can read from the Redpanda topic, using the xref:components:inputs/kafka.adoc[`kafka`] input:
+
+[source,yaml]
+----
+input:
+  kafka:
+ifndef::env-cloud[]
+    addresses: [ "${REDPANDA_CLUSTER}" ]
+endif::[]
+ifdef::env-cloud[]
+    addresses: [ "TODO" ]
+endif::[]
+    topics: [ articles ]
+    consumer_group: rp_connect_articles_group
+    tls:
+      enabled: true
+    sasl:
+      mechanism: SCRAM-SHA-256
+ifndef::env-cloud[]
+      user: "${REDPANDA_USER}"
+      password: "${REDPANDA_PASSWORD}"
+endif::[]
+ifdef::env-cloud[]
+      user: "TODO"
+      password: "TODO"
+endif::[]
+----
+
+Use https://ollama.com/library/nomic-embed-text[Nomic Embed] to compute embeddings. Since each request only applies to a single document, you can scale this by making requests in parallel across document batches.
```

Comment:
```suggestion
Use https://ollama.com/library/nomic-embed-text[Nomic Embed^] to compute embeddings. Since each request only applies to a single document, you can scale this by making requests in parallel across document batches.
```